src/main/java/uk/gov/justice/hmpps/prison/service/OffenderLocationService.kt

```diff
@@ -4,29 +4,54 @@ import com.fasterxml.jackson.annotation.JsonIgnore
 import com.fasterxml.jackson.annotation.JsonInclude
 import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
 import io.swagger.v3.oas.annotations.media.Schema
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 import uk.gov.justice.hmpps.prison.api.model.OffenderSummary
 import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyInternalLocation
 import uk.gov.justice.hmpps.prison.repository.jpa.model.LivingUnitReferenceCode
 import uk.gov.justice.hmpps.prison.repository.jpa.repository.AgencyInternalLocationRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.BedAssignmentHistoriesRepository
 import uk.gov.justice.hmpps.prison.security.VerifyBookingAccess
 
 @Service
 @Transactional(readOnly = true)
 class OffenderLocationService(
   private val agencyInternalLocationRepository: AgencyInternalLocationRepository,
+  private val bedAssignmentHistoriesRepository: BedAssignmentHistoriesRepository,
 ) {
   @VerifyBookingAccess(overrideRoles = ["SYSTEM_USER", "GLOBAL_SEARCH", "VIEW_PRISONER_DATA"])
   fun getOffenderLocation(@Suppress("UNUSED_PARAMETER") bookingId: Long, summary: OffenderSummary): OffenderLocation {
     // booking not active or no current internal location then no location to report
     if (summary.currentlyInPrison != "Y" || summary.internalLocationId == null) return OffenderLocation()
 
-    val location = agencyInternalLocationRepository.findOneByLocationId(summary.internalLocationId.toLong())
+    val currentLocationId = summary.internalLocationId.toLong()
+    val location = agencyInternalLocationRepository.findOneByLocationId(currentLocationId)
       .orElseThrow { RuntimeException() }
 
-    return OffenderLocation(levels = location.getLocationWithParents())
+    val lastPermanentLocation = if (isTemporaryLocation(location.locationCode)) {
+      val cells = bedAssignmentHistoriesRepository.findAllByBedAssignmentHistoryPKOffenderBookingId(
+        bookingId,
+        PageRequest.of(0, 10, Sort.by("bedAssignmentHistoryPK.sequence").descending()),
```

Comment:
we assume that they won't have moved between 10 different temporary locations so only get the last 10

Reply:
That's the same logic I used also,.. still a little uneasy as it's an arbitrary number.. But I do think this can be changed at a later date if need be.